app/controllers/admin/facilities_controller.rb

```diff
@@ -119,13 +119,21 @@ def validate_duplicate_rows
         facilities_slice.count != facilities_slice.uniq.count
   end
 
+  def valid_facility_size?(facility)
+    facility_size = facility[:facility_size]
+    facility_size.blank? || Facility::facility_sizes.include?(facility_size)
+  end
+
   def validate_facilities
     @row_errors = []
     row_num = 2
     @facilities.each do |facility|
-      import_facility = Facility.new(facility)
-      @row_errors << [row_num, import_facility.errors.full_messages.to_sentence] if
-          import_facility.invalid?
+      if valid_facility_size?(facility)
+        import_facility = Facility.new(facility)
+        @row_errors << [row_num, import_facility.errors.full_messages.to_sentence] if import_facility.invalid?
+      else
+        @errors << "Facility size on row #{row_num} is invalid"
+      end
```

Comment:
Is there a better way to validate `facility_size`? Looks like rails does not support `enum` inclusion validations and just throws an `ArgumentError` when `Facility.new` is created. From https://github.com/rails/rails/issues/13971 it seems like doing a separate check in the controller is the most reasonable

Reply:
Yeah I've noticed our usage of enum fields doesn't work well w/ our primarily API driven workflows. 

Want to give this [LiberalEnum](https://github.com/rails/rails/issues/13971#issuecomment-542757073) patch a try ? That would be useful for several of our enum types we want to be able to validate.